Automattic/pocket-casts-ios
# Settings Sync: `new_settings_storage` Feature Flag

Adds a `new_settings_storage` feature flag for migrating settings to new JSON based format.

The original `settings_sync` flag was trying to accomplish two things:
1. Ensure we can fall back to the same old settings logic in case something breaks
2. Ensure we can enable/disable syncing between devices.

This separates the settings storage changes from the syncing changes. This way, we can disable syncing without disabling storage changes, retaining any modified dates as long as the `new_settings_storage` flag is enabled.

This fixes issues with settings being dropped by simply disabling the syncing logic instead of reverting to the old user defaults. The behavior now matches Android.

I feel that the new flag is still useful to have in case something gets messed up with settings conversion, we have a way to back out ALL of the storage related changes _just in case_.

## To test

* Install this branch on two devices/simulators
* Enable the `new_settings_storage` AND `settings_sync` feature flags & restart the app (this restart won't be needed after https://github.com/Automattic/pocket-casts-ios/pull/1506)
* D1: Change a setting and refresh
* D2: Refresh and ensure setting is synced
* D1: Disable `settings_sync` flag + restart
* D1: Change a setting and refresh
* D2: Refresh and note that the setting remains unchanged
* D2: Change a setting and refresh
* D1: Enable `settings_sync` flag + restart
* D1: Refresh to be sure settings are synced
* D2: Refresh and notice that changes made while `settings_sync` was disabled are reflected now 

## Checklist
